blocks/identity-block/components/HeadlinedSubmitForm/index.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useRef } from 'react';
+import PropTypes from '@arc-fusion/prop-types';
+
+import { PrimaryFont } from '@wpmedia/shared-styles';
+import Button, { BUTTON_STYLES, BUTTON_SIZES } from '../Button';
+
+import './styles.scss';
+
+const HeadlinedSubmitForm = ({
+  children,
+  headline,
+  buttonLabel,
+  formErrorText,
+  onSubmit = () => {},
+}) => {
+  const formRef = useRef();
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    const valid = formRef.current.checkValidity();
+    if (valid) {
+      const namedFields = Array.from(formRef.current.elements)
+        .filter((element) => element?.name && typeof element?.name !== 'undefined')
+        .reduce((accumulator, element) => ({
+          ...accumulator,
+          [element.name]: element.value,
+        }), {});
+      onSubmit(namedFields);
+    }
+  };
+
+  return (
+    <section>
+      <PrimaryFont
+        as="h2"
+        className="xpmedia-form-headline"
+      >
+        {headline}
+      </PrimaryFont>
+      <form
+        aria-label={headline}
+        className="xpmedia-form-submittable"
+        onSubmit={handleSubmit}
+        ref={formRef}
+      >
+        {children}
+        <Button
```

Comment:
Button type needs to be `submit`

Reply:
submit is the default when the button is in a form.

Reply:
That is only true if the value of the type is not present of invalid - the value being set is `type=button`

Reply:
Added to #1069